### PR TITLE
feat: SER-2514 Refactor settlement flow to use TNM reference number as the transaction id

### PR DESCRIPTION
**Pull request by amydoxym**
## Description

Refactor settlement flow to use TNM reference number as the transaction id

### CODE REVIEW

**Review by gilclefes — APPROVED** [review]
looks good to me

File: src/main/java/org/mifos/connector/ams/fineract/camel/route/FineractRouteBuilder.java [hunk 1]
```diff
@@ -172,7 +172,17 @@ public void configure() throws Exception {
...
+                        Object useWorkflowIdAsTransactionIdObj = channelRequest.get("useWorkflowIdAsTransactionId");
                         String transactionId = exchange.getProperty(TRANSACTION_ID, String.class);
+                        // TNM flow: since the validation flow uses workflowId in the AMS validation call, we need to
+                        // use it here as well
+                        if (useWorkflowIdAsTransactionIdObj != null && (Boolean) useWorkflowIdAsTransactionIdObj
+                                && channelRequest.has("workflowId")
+                                && channelRequest.get("workflowId") instanceof String workflowId) {
+                            transactionId = workflowId;
+                        }
+
```
Comment (victorjbassey) [inline src/main/java/org/mifos/connector/ams/fineract/camel/route/FineractRouteBuilder.java:185]: isn't it possible to have the TNM connector set the correct transaction ID that will then be used here? 
It looks like this logic should belong in the TNM connector

Comment (amydoxym) [reply src/main/java/org/mifos/connector/ams/fineract/camel/route/FineractRouteBuilder.java:185]: Good point! so the thing here is that we want to give a possibility to do search in Ops app with the TNM reference number since it will be the value that makes more sense to them. At the time of validation, we don't have it, we just have our generated reference. So this change is to set the TNM reference as the transactionId (which is what we use to search in Ops app) and do this tweak when we are calling the Fineract confirmation API (for Roster, it is already using the external id when calling the confirmation API).

In short, we  want to set the TNM reference as transactionId but Fineract at the time of confirmation don't know about the TNM ref.  So we just provide to Fineract what it knows and for the rest, keep using TNM ref. Hope you get the point. If not, we can have a quick call to discuss it

**Review by victorjbassey — APPROVED** [review]
